### PR TITLE
PWN-4263 - onboarding screen with no logic added

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4263

## Description of Work

Added New Onboarding screen with slider

## Screenshots (Optional)

Android Emulator - Pixel_5_API_30:5554 2022-07-21 13-10-29

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/auth/AuthModule.kt [hunk 1]
```diff
@@ -38,6 +42,11 @@ object AuthModule {
     fun create() = module {
         single { BiometricManager.from(androidContext()) }
 
+        factory { GoogleSignInHelper() }
+        factory { WalletWeb3AuthManager(get(), get(), get(), get()) }
+
+        factory { NewOnboardingPresenter(get()) } bind NewOnboardingContract.Presenter::class
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/AuthModule.kt:48]: i would extract it as an extension Module.newOnboardingScreensModule inside AuthModule

File: app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            R.string.onboarding_slide_1_text,
+        ).toBundle()
+    }
+
+    private val signInHelper: GoogleSignInHelper by inject()
+
+    private val googleSignInLauncher = registerForActivityResult(
+        ActivityResultContracts.StartIntentSenderForResult()
+    ) {
+        handleSignResult(it)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt:45]: can be method reference

File: app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                presenter.onSignUpButtonClicked()
+            }
+        }
+
+        requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner) {
+            popBackStack()
+        }
+    }
+
+    override fun startGoogleFlow() = signInHelper.showSignInDialog(requireContext(), googleSignInLauncher)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt:65]: if the return type is Unit - than it should be block body

File: app/src/main/java/org/p2p/wallet/auth/ui/onboarding/SliderFragment.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        arguments?.let { args ->
+            val sliderFragmentArgs = SliderFragmentArgs.fromBundle(args)
+            with(binding) {
+                sliderIcon.setImageResource(sliderFragmentArgs.iconRes)
+                sliderTitle.setText(sliderFragmentArgs.slideTitle)
+                sliderText.setText(sliderFragmentArgs.slideText)
+            }
+        } ?: Timber.e("Error on getting SliderFragmentArgs")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/onboarding/SliderFragment.kt:28]: pass on `arguments` here please in one way or another

File: app/src/main/java/org/p2p/wallet/auth/ui/onboarding/SliderFragmentArgs.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+private const val ICON_ARG = "sliderIcon"
+private const val TITLE_ARG = "sliderTitle"
+private const val TEXT_ARG = "sliderText"
+
+data class SliderFragmentArgs(
+    @DrawableRes val iconRes: Int,
+    @StringRes val slideTitle: Int,
+    @StringRes val slideText: Int,
+) {
+    fun toBundle(): Bundle = Bundle().apply {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/onboarding/SliderFragmentArgs.kt:17]: can be bundleOf

File: app/src/main/java/org/p2p/wallet/utils/BundleExtensions.kt [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+fun <T : Parcelable> Bundle.requireParcelable(key: String): T {
+    return if (containsKey(key)) {
+        requireNotNull(getParcelable(key))
+    } else {
+        throw IllegalArgumentException("Required argument \"$key\" is missing!")
+    }
+}
+
+fun Bundle.requireString(key: String): String {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/utils/BundleExtensions.kt:15]: i saw that the naming style for this type of extensions is like:

1. getOrThrow / getOrNull / getOrDefault
so i suggest to use getOrThrow

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -152,6 +152,16 @@
     <string name="auth_fingerprint_login">Fingerprint login</string>
     <string name="auth_fingerprint_login_message">Use your fingerprint for faster easier access to your wallet</string>
 
+    <!--Onboarding-->
+    <string name="onboarding_create_text">Create a new wallet</string>
```
Comment (gslevinkov) [inline app/src/main/res/values/strings.xml:156]: let's extract onboarding texts to separate strings file shall we?

File: ui-kit/src/main/res/values/white_palette.xml [hunk 8]
```diff
@@ -12,12 +12,14 @@
     <color name="color_green">#2DB533</color>
     <color name="color_green_transparent">#802DB533</color>
 
+    <color name="night">#2B2B2B</color>
+    <color name="night_60">#992B2B2B</color>
```
Comment (gslevinkov) [inline ui-kit/src/main/res/values/white_palette.xml:16]: new colors in uikit??

**Review by gslevinkov — APPROVED** [review]

File: ui-kit/src/main/res/values/white_palette.xml (hunk 8 above)
Comment (kabramovp2p) [reply ui-kit/src/main/res/values/white_palette.xml:16]: we do not have alpha but we use it in UI :(

**Review by kuznetsov — APPROVED** [review]

**Pull request merged**